WPMedia/fusion-news-theme-blocks
# Tmedia 731 tags block

## Description

Update the tags block to v2 architecture

## Jira Ticket

- [TMEDIA-731](https://arcpublishing.atlassian.net/browse/TMEDIA-731) sub-task of [TMEDIA-713](https://arcpublishing.atlassian.net/jira/software/c/projects/TMEDIA/boards/875?modal=detail&selectedIssue=TMEDIA-713&assignee=5e387d6a1b1d910e5dfd7a9b)

## Acceptance Criteria

Tags Bar block is migrated to the new architecture.

Includes Custom fields of:

‘Lazy Load block?’ checkbox with help text

## Test Steps

1. Checkout this branch `git checkout TMEDIA-713-content-item-blocks`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/article-tag-block`
3. Go to http://localhost/pf/content-item-blocks/?_website=the-gazette. Compare with https://corecomponents-the-gazette-prod.cdn.arcpublishing.com/content-item-blocks/?_website=the-gazette and make sure that the gap matches the use of spacing before. 

## Effect Of Changes

### Before

- Matches v1 design

### After

- Pill font size matches the style designated in figma 

## Dependencies or Side Effects

- Discrepancies in font size are listed in discrepancies https://arcpublishing.atlassian.net/wiki/spaces/TI/pages/3138945094/Blocks+Design+Discrepancies
- feature pack https://github.com/WPMedia/arc-themes-feature-pack/pull/261
- manifest https://github.com/WPMedia/arc-themes-manifests/pull/61

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

        please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._


# Question

- I should be able to override the flex-direction in the theme token file right? I feel like I'm missing something here.  -> So those default inline styles can't be overriden by the theme file. Instead, styles applied to those divs should use the inline styles.
<img width="1202" alt="Screen Shot 2022-05-13 at 16 55 12" src="https://user-images.githubusercontent.com/5950956/168395516-9354d24f-7d55-4a02-893c-f519117fd997.png">
- We're using line-height to establish height on the pill. If there's no text then there's no height. This is a change. Is this an ok change? Would we expect a different behavior if there's no text, like not rendering anything? https://www.chromatic.com/test?appId=5f91e4721ccaba0022a10782&id=6287b18eab872a003a89907a
- I removed the top and bottom spacing for the block itself. I also removed the border. Both of these should be handled in the future by a layout, right? https://www.chromatic.com/test?appId=5f91e4721ccaba0022a10782&id=6287b18eab872a003a899072
